### DISCUSSION TITLE
Mismatched Cube in Blocks Environment

### DISCUSSION THREAD

**Discussion by Gastastrophe** [microsoft/AirSim#4278] [Support Q&A]
So I was taking some segmentation images in the blocks environment and using them to guide my drone, and I noticed some odd behaviors. I had set a custom obstacle that I added to the environment to black, and the drone is supposed to move towards the center of the obstacle in the image. However, I noticed there was a sliver of black on the side of the group8 cube stack that was misleading my algorithm (See image). I looked into it, and one specific block (The first one in the group) in this stack is a normal cube instead of a "rounded" cube. Is there a reason for this? Can someone confirm this is true for the latest build of Airsim, since my version of the environment is an older one that I have been working with for a while, and no longer matches my build of Airsim? This is fairly inconvenient for taking Segmentation images, and I see no actual benefit to it.

![Mismatch](https://user-images.githubusercontent.com/56171708/148669899-995aa879-7268-4397-b6ca-b12033278e60.jpg)
h

**Comment by zimmy87** [comment]
I can confirm what you're seeing still exists in the latest build of AirSim. I don't have any context for why this one cube would be different from the others, I suspect it was just overlooked. If you want to keep using the same version of your environment, you can call the `simSetSegmentationObjectID` API with mesh_name `TemplateCube_48` to overwrite the segmentation ID.

**Comment by zimmy87** [comment]
I put out PR #4331 to fix this
